libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1455,8 +1456,10 @@ void AP_OSD_Screen::draw_horizon(uint8_t x, uint8_t y)
 {
     AP_AHRS &ahrs = AP::ahrs();
     WITH_SEMAPHORE(ahrs.get_semaphore());
```

Comment:
Were still taking the ahrs semaphore, we do use the ahrs in the new function. I wonder if we should take the vehicle scheduler-semaphore instead. 

I'm not really sure what we should be doing....